core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1156,6 +1156,18 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * Execute a block of code that evaluates to Unit, stop SparkContext is any uncaught exception
```

Comment:
Add a comment contrasting this to tryOrExit, saying that this method is suitable for the driver while tryOrExit should be used for other JVMs started by Spark, over which we have full control. Also, second part should say something like "stopping the SparkContext if there is any uncaught exception."